### ISSUE TITLE
cast call token parsing error

### ISSUE THREAD

**Issue by brianmcmichael** [foundry-rs/foundry#1588]
### Component

Cast

### Have you ensured that all of these are up to date?

- [X] Foundry
- [X] Foundryup

### What version of Foundry are you on?

forge 0.2.0 (752dd63 2022-05-10T00:03:36.474260356+00:00)

### What command(s) is the bug in?

cast call

### Operating System

Linux

### Describe the bug

Getting an error when attempting to retrieve a string from a remote contract.

Expected behavior (ref. `seth`):

```haskell
$ seth call 0x5a464c28d19848f44199d003bef5ecc87d090f87 "name(bytes32)(string)" 0x5257413030312d410000000000000000000000000000000000000000000000
> RWA001-A: 6s Capital
```

Actual behavior in `cast`:

```haskell
$ cast call 0x5a464c28d19848f44199d003bef5ecc87d090f87 "name(bytes32)(string)" 0x5257413030312d410000000000000000000000000000000000000000000000
> Error: 
>    0: Failed to parse tokens
>    1: Invalid data
> 
> Location:
>    /home/runner/work/foundry/foundry/utils/src/lib.rs:531
>
> Backtrace omitted. Run with RUST_BACKTRACE=1 environment variable to display it.
> Run with RUST_BACKTRACE=full to include source snippets.
```

**Comment by dmfxyz** [comment]
This is because the arg is not padded to 32 bytes. 
```sh
$> echo -n 5257413030312d410000000000000000000000000000000000000000000000 | wc -c`
62
```

If you pad properly to 32 bytes, the call works:
```sh
cast call 0x5a464c28d19848f44199d003bef5ecc87d090f87 "name(bytes32)(string)" 0x5257413030312d41000000000000000000000000000000000000000000000000
RWA001-A: 6s Capital
```
Seth is padding for you. @onbjerg @gakonst @mattsse does Foundry want to support this padding>

**Comment by gakonst** [comment]
We want to support padding  yes

**Comment by dmfxyz** [comment]
cool. I can take this on tomorrow if no one objects.

**Comment by gakonst** [comment]
yep go for it @dmfxyz

**Comment by gakonst** [comment]
fixed in #1594
